nixos/modules/hardware/opengl.nix

```
@@ -125,14 +125,10 @@ in
 
   config = mkIf cfg.enable {
 
-    assertions = [
-      { assertion = cfg.driSupport32Bit -> pkgs.stdenv.isx86_64;
-        message = "Option driSupport32Bit only makes sense on a 64-bit system.";
-      }
-      { assertion = cfg.driSupport32Bit -> (config.boot.kernelPackages.kernel.features.ia32Emulation or false);
-        message = "Option driSupport32Bit requires a kernel that supports 32bit emulation";
-      }
-    ];
+    system.requiredKernelConfig =
+      if cfg.driSupport32Bit
+      then map config.lib.kernelConfig.isYes [ "IA32_EMULATION" ]
+      else null;
```

Did you test this? The option accepts a list, `null` shouldn't work.
Anyway, I suggest

```suggestion
    system.requiredKernelConfig =
      lib.optional cfg.driSupport32Bit 
      (config.lib.kernelConfig.isYes "IA32_EMULATION")
```

---

It did work (surprisingly), but I wasn't very proud of it -- I'm new to Nix. This seems much better.